## Dataset Information : MINST
#### Description :
- Fashion training set consists of 70,000 images divided into 60,000 training and 10,000 testing samples. Dataset sample consists of 28x28 grayscale image, associated with a label from 10 classes. 

- The 10 classes are as follows:  
- 0 => T-shirt/top
- 1 => Trouser
- 2 => Pullover
- 3 => Dress
- 4 => Coat
- 5 => Sandal
- 6 => Shirt
- 7 => Sneaker
- 8 => Bag
- 9 => Ankle boot

Each image is 28 pixels in height and 28 pixels in width, for a total of 784 pixels in total. Each pixel has a single pixel-value associated with it, indicating the lightness or darkness of that pixel, with higher numbers meaning darker. This pixel-value is an integer between 0 and 255. 

![image.png](attachment:image.png)

In [1]:
## Importing necessary lib
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import random
from scipy import stats
import warnings
warnings.filterwarnings("ignore")
plt.style.use('dark_background')


### Data Preprocessing, Data Visualisation and Data Analysis


In [3]:
## Load the data
df_train_fash = pd.read_csv(r"F:\Datasets\fashion-mnist_train.csv", sep = ',')
df_test_fash = pd.read_csv(r"F:\Datasets\fashion-mnist_test.csv", sep = ',')

In [4]:
df_train_fash.head()

label  pixel1  pixel2  pixel3  pixel4  pixel5  pixel6  pixel7  pixel8  \
0      2       0       0       0       0       0       0       0       0   
1      9       0       0       0       0       0       0       0       0   
2      6       0       0       0       0       0       0       0       5   
3      0       0       0       0       1       2       0       0       0   
4      3       0       0       0       0       0       0       0       0   

   pixel9  ...  pixel775  pixel776  pixel777  pixel778  pixel779  pixel780  \
0       0  ...         0         0         0         0         0         0   
1       0  ...         0         0         0         0         0         0   
2       0  ...         0         0         0        30        43         0   
3       0  ...         3         0         0         0         0         1   
4       0  ...         0         0         0         0         0         0   

   pixel781  pixel782  pixel783  pixel784  
0         0         0         0         0  
1         0         0         0         0  
2         0         0         0         0  
3         0         0         0         0  
4         0         0         0         0  

[5 rows x 785 columns]

In [5]:
df_train_fash.shape

(60000, 785)

In [6]:
df_train_fash.isnull().sum()

label       0
pixel1      0
pixel2      0
pixel3      0
pixel4      0
           ..
pixel780    0
pixel781    0
pixel782    0
pixel783    0
pixel784    0
Length: 785, dtype: int64

In [7]:
df_train_fash.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60000 entries, 0 to 59999
Columns: 785 entries, label to pixel784
dtypes: int64(785)
memory usage: 359.3 MB


In [8]:
df_train_fash.describe().T

count       mean        std  min  25%  50%  75%    max
label     60000.0   4.500000   2.872305  0.0  2.0  4.5  7.0    9.0
pixel1    60000.0   0.000900   0.094689  0.0  0.0  0.0  0.0   16.0
pixel2    60000.0   0.006150   0.271011  0.0  0.0  0.0  0.0   36.0
pixel3    60000.0   0.035333   1.222324  0.0  0.0  0.0  0.0  226.0
pixel4    60000.0   0.101933   2.452871  0.0  0.0  0.0  0.0  164.0
...           ...        ...        ...  ...  ...  ...  ...    ...
pixel780  60000.0  17.911483  45.149388  0.0  0.0  0.0  0.0  255.0
pixel781  60000.0   8.520633  29.614859  0.0  0.0  0.0  0.0  255.0
pixel782  60000.0   2.753300  17.397652  0.0  0.0  0.0  0.0  255.0
pixel783  60000.0   0.855517   9.356960  0.0  0.0  0.0  0.0  255.0
pixel784  60000.0   0.070250   2.125870  0.0  0.0  0.0  0.0  170.0

[785 rows x 8 columns]

In [9]:
df_train_fash['label'].nunique()

10

In [10]:
df_train_fash['label'].value_counts()

2    6000
9    6000
6    6000
0    6000
3    6000
4    6000
5    6000
8    6000
7    6000
1    6000
Name: label, dtype: int64

### Obs : 
- Training data having order 60000*785
- There are 10 class are present and all classes are fully balanced
- Few features may have contain outlier
- There is no missing value present in any of the feature